src/OpenTelemetry/Trace/ActivityProcessor.cs

```diff
@@ -28,13 +28,29 @@ public abstract class ActivityProcessor
         /// Activity start hook.
         /// </summary>
         /// <param name="activity">Instance of activity to process.</param>
-        public abstract void OnStart(Activity activity);
+        public void OnStart(Activity activity)
+        {
+            if (Sdk.SuppressInstrumentation)
```

Comment:
Just wondering out loud here... should the `SuppressInstrumentation` check go in ActivityProcessor or should it be part of the ActivitySampler? If `SuppressInstrumentation == true` forced a lower sampling decision (None or PropagationOnly) that would be better for perf I think.